subsys/rtio/rtio_workq.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024 Croxel Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/rtio/work.h>
+#include <zephyr/kernel.h>
+
+#define RTIO_WORKQ_PRIO_MED		CONFIG_RTIO_WORKQ_PRIO_MED
+#define RTIO_WORKQ_PRIO_HIGH		RTIO_WORKQ_PRIO_MED - 1
+#define RTIO_WORKQ_PRIO_LOW		RTIO_WORKQ_PRIO_MED + 1
+
+K_P4WQ_DEFINE(rtio_workq,
+	      CONFIG_RTIO_WORKQ_THREADS_POOL,
+	      CONFIG_RTIO_WORKQ_STACK_SIZE);
+
+K_MEM_SLAB_DEFINE_STATIC(rtio_work_items_slab,
+			 sizeof(struct rtio_work_req),
+			 CONFIG_RTIO_WORKQ_POOL_ITEMS,
+			 4);
+
+static void rtio_work_handler(struct k_p4wq_work *work)
+{
+	struct rtio_work_req *req = CONTAINER_OF(work,
+						 struct rtio_work_req,
+						 work);
+	const struct device *dev = req->dev;
+	struct rtio_iodev_sqe *iodev_sqe = req->iodev_sqe;
+
+	req->handler(dev, iodev_sqe);
```

Comment:
The handler shouldn't need device, the iodev_sqe has a link to the iodev which should link to the device much like it does already in submit

Reply:
You're right - I've scrubbed this from the APIs. Each handler should know how to retrieve its own device reference (e.g: sensors do it through `sensor_read_config`, SPI through `spi_dt_spec` and so forth).

Thanks!